File: Dynamic Programming Applications In Machine Learning and Genomics/Length of a Longest Path in a Grid/longest_path_grid.py
#python3
import sys

def longest_path(n,m,down,right):
    # TODO: your code here
    return -1

if __name__ == "__main__":
    n,m = map(int, sys.stdin.readline().strip().split())
    down = [list(map(int, sys.stdin.readline().strip().split()))
            for _ in range(n)]
    sys.stdin.readline()
    right = [list(map(int, sys.stdin.readline().strip().split()))
            for _ in range(n+1)]

    print(longest_path(n,m,down,right))

File: Dynamic Programming Applications In Machine Learning and Genomics/Length of a Longest Path in a Grid/LongestPathGrid.java
import java.util.Scanner;

public class LongestPathGrid {
    private static int longestPath(int n, int m, int[][] down, int[][] right) {
        // TODO: your code here
        return -1;
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        int n = stdin.nextInt();
        int m = stdin.nextInt();
        stdin.nextLine();

        int[][] down = new int[n][m+1];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m+1; j++) {
                down[i][j] = stdin.nextInt();
            }
            stdin.nextLine();
        }

        stdin.nextLine();

        int[][] right = new int[n+1][m];
        for (int i = 0; i < n+1; i++) {
            for (int j = 0; j < m; j++) {
                right[i][j] = stdin.nextInt();
            }
            stdin.nextLine();
        }

        System.out.println(longestPath(n, m, down, right));
    }
}
